service/cluster/cluster.go

```diff
@@ -402,6 +404,8 @@ func (s *clusterImpl) create(req *v1.CreateClusterRequest, owner, eventID string
 		"cluster-owner", owner,
 	)
 
+	metrics.FlavorsUsedCounter.WithLabelValues(flav.ID).Inc()
```

Comment:
IMHO this should be moved to after the `s.argoWorkflowsClient.CreateWorkflow()` call, where we handle the other post-workflow creation actions, like logging, BigQuery integration.